gnome-image-installer/pages/install/gis-install-page.c

```diff
@@ -433,7 +494,21 @@ gis_install_page_copy (GisPage *page)
         }
     }
 
+  priv->writing = FALSE;
+
 out:
+
+  /*
+   * If there's a message dialog asking whether the user wants to quit, and
+   * we finished writing, hide and ignore that dialog.
+   */
+  if (priv->warning_dialog)
+    {
+      gtk_dialog_response (GTK_DIALOG (priv->warning_dialog), GTK_RESPONSE_CANCEL);
```

Comment:
Are you sure it's safe to do this from a non-main thread? Shouldn't this go into `gis_install_page_teardown`?